### PR TITLE
Add UserDao

### PR DESCRIPTION
**Pull request by Arturnix**

### CODE REVIEW

File: src/main/java/pl/arturzgodka/datamodel/User.java [hunk 1]
```diff
@@ -12,10 +16,20 @@ public class User {
...
+    @NotEmpty (message = "Email may not be empty")
+    @NotBlank (message = "Email may not be blank")
     private String email;
+    @NotNull (message = "Password may not be null")
+    @NotEmpty (message = "Password may not be empty")
+    @NotBlank (message = "Password may not be blank")
     private String password;
-    @OneToMany //1 user ma wiele postaci.
+    @OneToMany //1 user ma wiele postaci przypisanych do BattleTag - konta.
     List<Character> characters;
```
Comment (blazejprzyluski) [inline src/main/java/pl/arturzgodka/datamodel/User.java:29]: private

File: src/main/java/pl/arturzgodka/databaseutils/UserDao.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        user.setPassword(passwordEncoder.encode(user.getPassword())); //aby nie trzymac hasel jako plain text w bazie danych.
+        session.merge(user);
+        t.commit();
+        session.close();
+    }
+
+    public void deleteUser(String email) {
+        Session session = sessionFactory.openSession();
+        Transaction transaction = session.beginTransaction();
+        User user = findUserByEmail(email);
```
Comment (blazejprzyluski) [inline src/main/java/pl/arturzgodka/databaseutils/UserDao.java:28]: what if user is not found?

**Review by blazejprzyluski — COMMENTED** [review]
Tests please

File: src/main/java/pl/arturzgodka/datamodel/User.java (hunk 1 above)
Comment (Arturnix) [reply src/main/java/pl/arturzgodka/datamodel/User.java]: Done

**Comment by Arturnix** [comment]
Tests added

**Pull request merged**